### DISCUSSION TITLE
Automate chargin

### DISCUSSION THREAD

**Discussion by PStenbumling** [Hyundai-Kia-Connect/kia_uvo#714] [Q&A]
Hi!

Unfortunately my charger does not have api capabillities yet. (Huawei).
So when connecting the charger to the car (Niro EV My23) the charging start immediately.

So my question is, how can i disable the charging and instead configure a schedule based on electricity prices.
Could perhaps trigger on event and start charging.

**Comment by DJBenson** [comment]
To stop the car automatically charging, set you peak and off peak times on the head unit (maybe also in the app) and ensure you select the option to only charge off peak - if you set the off peak period to 1 minute in 24 hours it will have the effect of stopping the car automatically charging when it's plugged in (unless you hit that exact minute in the day 😛).

NOTE: for both my previous car (e-Niro) and current (EV6) the peak/off peak times are lost REGULARLY which is infuriating if you don't spot the times have been reset to 00:00:00.

I use Home Assistant, the Octopus Energy integration and the Kia integration to control charging of my car down to 30 minute blocks so that it only charges when the unit rate is at or below a threshold OR I have forced the charge.

Here's my EV dashboard - the section on the right is the custom automation for matching charging to unit rates;
<img width="1498" alt="image" src="https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/1013909/6c344d39-42ec-45a4-8ce5-0a42750bade0">

If I toggle "Specific Range Required" then I get this and the unit rate is ignored (useful for when I have a long journey which I know I can't achieve optimal charging for);
<img width="507" alt="image" src="https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/1013909/5443930d-25f7-487c-bbea-31113a40cfd2">
